### ISSUE TITLE
Members added via Add-Type not showing up in VSCode Intellisense

### ISSUE THREAD

**Issue by netcorefan1** [PowerShell/PowerShell#19711]
### Prerequisites

- [X] Write a descriptive title.
- [X] Make sure you are able to repro it on the [latest released version](https://github.com/PowerShell/PowerShell/releases)
- [X] Search the existing issues.
- [X] Refer to the [FAQ](https://github.com/PowerShell/PowerShell/blob/master/docs/FAQ.md).
- [X] Refer to [Differences between Windows PowerShell 5.1 and PowerShell](https://docs.microsoft.com/powershell/scripting/whats-new/differences-from-windows-powershell).

### Steps to reproduce

This is the same issue I reported [here.](https://github.com/PowerShell/vscode-powershell/issues/4585). It has been confirmed to be a PowerShell issue, and not a VSCode issue.

Members added via Add-Type are available in the terminal via tab completion, but are completely absent in VSCode Intellisense.
Whether the types are placed in a profile or in a manifest, it doesn't make any difference.

To reproduce:
```
Update-TypeData -TypeName System.String -MemberType ScriptMethod -MemberName printString  -Value {
    Write-Host "printString() printed $this"
}
```

### Expected behavior

```console
The same of the Terminal, `printString()` available and selectable:
`'999'. + TAB + pr`:

    > printString(
    > PadRight(
    > GetPinnableReference(
    > ToUpper(
    > CompareTo(
```
```


### Actual behavior

```console
VSCODE (method executes without any problem, but it does not show up in intellisense)
`'999'.pr`:

    > PadRight(
```
```


### Error details

_No response_

### Environment data

```powershell
Name                           Value
----                           -----
PSVersion                      7.3.4
PSEdition                      Core
GitCommitId                    7.3.4
OS                             Microsoft Windows 10.0.22621
Platform                       Win32NT
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0…}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0
```

**Comment by mklement0** [comment]
Just to describe the problem in more detail:

* Fundamentally, for such an ETS member to be recognized by IntelliSense, the `Update-TypeData` call must have actually _executed_, i.e. the script must have been run / debugged in a session at least once.

* However, this does _not_ work if you have the `PowerShell › Debugging: Create Temporary Integrated Console` setting turned on (which is definitely advisable), because it is then only the temporary session, not the one underlying the editor, that sees the new member.

**Comment by SeeminglyScience** [comment]
> * Fundamentally, for such an ETS member to be recognized by IntelliSense, the `Update-TypeData` call must have actually _executed_, i.e. the script must have been run / debugged in a session at least once.

Yeah that's correct. @netcorefan1 you mentioned this was in your profile, note that unless you're using `$profile.CurrentUserAllHosts` (which is `profile.ps1`) the profile name will change per host. Either way you should be able to highlight the lines, press F8 and it will be present in intellisense afterwards.

Also note that `Add-Type` and `Update-TypeData` are two very different commands. I'm going to edit the issue title to reflect that, but if you were referring to `Add-Type` as well please add an example (Though, the same rules apply, it must be run first. It cannot be statically inferred).

**Comment by netcorefan1** [comment]
> Just to describe the problem in more detail:
> 
>     * Fundamentally, for such an ETS member to be recognized by IntelliSense, the `Update-TypeData` call must have actually _executed_, i.e. the script must have been run / debugged in a session at least once.
> 
>     * However, this does _not_ work if you have the `PowerShell › Debugging: Create Temporary Integrated Console` setting turned on (which is definitely advisable), because it is then only the temporary session, not the one underlying the editor, that sees the new member.


Thanks @mklement0 ! It worked once I unchecked `PowerShell › Debugging: Create Temporary Integrated Console` after the first call to  `Update-TypeData`. However I remember that I enabled temporary integrated console because I was experiencing several annoyances during development. If I remember well, one of them was the fact that without a temporary console, I was forced to restart VSCode each time at each minor change to the code and this was making development too much frustrating. 

Having intellisense working is something that I cannot renounce, so I will leave this unchecked, but how I can solve such problems?

**Comment by netcorefan1** [comment]
> Also note that `Add-Type` and `Update-TypeData` are two very different commands. I'm going to edit the issue title to reflect that, but if you were referring to `Add-Type` as well please add an example (Though, the same rules apply, it must be run first. It cannot be statically inferred).

Yes, I was referring to boths although `Update-TypeData` has been the last one I was playing with and forgot to update the title.

> > * Fundamentally, for such an ETS member to be recognized by IntelliSense, the `Update-TypeData` call must have actually _executed_, i.e. the script must have been run / debugged in a session at least once.
> 
> Yeah that's correct. @netcorefan1 you mentioned this was in your profile, note that unless you're using `$profile.CurrentUserAllHosts` (which is `profile.ps1`) the profile name will change per host. Either way you should be able to highlight the lines, press F8 and it will be present in intellisense afterwards.
> 


By restoring the default integrated console, intellisense started to work (no matter where my declaration are stored as long as they are run once), but by doing this I'm left with annoyances I stated in my previous comment.

I tried by storing the declaration in `C:\Users\User\Documents\PowerShell\Microsoft.PowerShell_profile.ps1`, but it didn't worked (is this because this is the wrong profile?).

I have been forced to create a `MyModule.psd1` manifest because in `MyModule.psm1` I was not able to access and reference types of external DLLs from within PS classes (`type not found error`). From what I know [this](https://stackoverflow.com/questions/42837447/powershell-unable-to-find-type-when-using-ps-5-classes) seems to be a well known problem in PS7. Many times has gone, so maybe that it has been fixed and there could be some mistake from my side.
Keeping custom types declaration (`add-type`,  `Update-TypeData` etc) inside `MyModule.psm1` (hopefully without recurring to manifests in the future) rather an external profile is what I prefer because this is the module where types belong to and they should be instantiated only if the module is needed and referenced.
Is this possible?
